### ISSUE TITLE
500 error when displaying issues

### ISSUE THREAD

**Issue by dloghin** [gitlabhq/gitlabhq#8479]
Hi,

I have recently updated my Gitlab from 5.4.0 to 7.5.0 using the manual installation steps and the migration guide from here:
http://lboynton.com/2013/01/29/how-to-migrate-gitlab-repositories-to-a-new-server/

Everything worked fine, except that I can't see the issues on my projects. I get "500 We're sorry, but something went wrong." Here is the error in log/production.log
```
Started GET "/gitlab/dashboard/issues" for 103.252.201.196 at 2014-12-23 12:28:39 +0800
Processing by DashboardController#issues as HTML
Completed 500 Internal Server Error in 268ms

ActionView::Template::Error (No route matches {:action=>"show", :controller=>"projects/issues", :format=>nil, :id=>#<Issue id: 7, title: "PiEstimator ArithmeticException", assignee_id: 4, author_id: 4, project_id: 16, created_at: "2014-11-16 06:04:27", updated_at: "2014-11-16 06:04:27", position: 0, branch_name: nil, description: "PiEstimator (default Java version in hadoop-1.2.1)...", milestone_id: nil, state: "opened", iid: nil>, :project_id=>#<Project id: 16, name: "hadoop-dev", path: "hadoop-dev", description: nil, created_at: "2014-01-28 06:44:15", updated_at: "2014-12-23 04:05:10", creator_id: 4, issues_enabled: true, wall_enabled: false, merge_requests_enabled: true, wiki_enabled: true, namespace_id: 4, issues_tracker: "gitlab", issues_tracker_id: nil, snippets_enabled: false, last_activity_at: "2014-11-25 05:30:59", import_url: nil, visibility_level: 0, archived: false, import_status: "none", repository_size: 192.93, star_count: 0>} missing required keys: [:id]):
    1: %li{ id: dom_id(issue), class: issue_css_classes(issue), url: project_issue_path(issue.project, issue) }
    2:   - if controller.controller_name == 'issues'
    3:     .issue-check
    4:       = check_box_tag dom_id(issue,"selected"), nil, false, 'data-id' => issue.id, class: "selected_issue", disabled: !can?(current_user, :modify_issue, issue)
  app/views/projects/issues/_issue.html.haml:1:in `_app_views_projects_issues__issue_html_haml__4442688756604332293_69838394065320'
  app/views/shared/_issues.html.haml:11:in `block (2 levels) in _app_views_shared__issues_html_haml__3262716708784517261_69838393121260'
  app/views/shared/_issues.html.haml:10:in `each'
  app/views/shared/_issues.html.haml:10:in `block in _app_views_shared__issues_html_haml__3262716708784517261_69838393121260'
  app/views/shared/_issues.html.haml:2:in `each'
  app/views/shared/_issues.html.haml:2:in `_app_views_shared__issues_html_haml__3262716708784517261_69838393121260'
  app/views/dashboard/issues.html.haml:14:in `_app_views_dashboard_issues_html_haml___2435821198925335946_69838389266100'
  app/controllers/dashboard_controller.rb:68:in `issues'
```
Since I suppose is a problem related to issue id, I checked the MyQSL issues table which seems to be ok:
```
mysql> SELECT id FROM issues;
+----+
| id |
+----+
|  1 |
|  2 |
|  3 |
|  4 |
|  5 |
|  6 |
|  7 |
+----+
7 rows in set (0.00 sec)
```

Here are the outputs of checks:
```
Checking Environment ...

Git configured for git user? ... yes

Checking Environment ... Finished

Checking GitLab Shell ...

GitLab Shell version >= 2.2.0 ? ... OK (2.2.0)
Repo base directory exists? ... yes
Repo base directory is a symlink? ... no
Repo base owned by git:git? ... yes
Repo base access is drwxrws---? ... yes
Satellites access is drwxr-x---? ... yes
hooks directories in repos are links: ... 
4/3 ... ok
4/14 ... ok
10/15 ... ok
4/16 ... ok
8/19 ... ok
4/20 ... ok
4/21 ... ok
8/22 ... ok
4/23 ... repository is empty
4/24 ... ok
4/25 ... ok
4/26 ... ok
Running /raid/git/gitlab/gitlab-shell/bin/check
Check GitLab API access: OK
Check directories and files: 
	/raid/git/repos/: OK
	/raid/git/.ssh/authorized_keys: OK
Test redis-cli executable: redis-cli 2.8.4
Send ping to redis server: PONG
gitlab-shell self-check successful

Checking GitLab Shell ... Finished

Checking Sidekiq ...

Running? ... yes
Number of Sidekiq processes ... 1

Checking Sidekiq ... Finished

Checking LDAP ...

LDAP is disabled in config/gitlab.yml

Checking LDAP ... Finished

Checking GitLab ...

Database config exists? ... yes
Database is SQLite ... no
All migrations up? ... yes
Database contains orphaned GroupMembers? ... no
GitLab config exists? ... yes
GitLab config outdated? ... no
Log directory writable? ... yes
Tmp directory writable? ... yes
Init script exists? ... yes
Init script up-to-date? ... yes
projects have namespace: ... 
4/3 ... yes
4/14 ... yes
10/15 ... yes
4/16 ... yes
8/19 ... yes
4/20 ... yes
4/21 ... yes
8/22 ... yes
4/23 ... yes
4/24 ... yes
4/25 ... yes
4/26 ... yes
Projects have satellites? ... 
4/3 ... yes
4/14 ... yes
10/15 ... yes
4/16 ... yes
8/19 ... yes
4/20 ... yes
4/21 ... yes
8/22 ... yes
4/23 ... yes
4/24 ... yes
4/25 ... yes
4/26 ... yes
Redis version >= 2.0.0? ... yes
Ruby version >= 2.0.0 ? ... yes (2.1.5)
Your git bin path is "/usr/bin/git"
Git version >= 1.7.10 ? ... yes (1.9.1)

Checking GitLab ... Finished
```
Environment
(for security reasons I replaced the name of my server)
```
System information
System:		Ubuntu 14.04
Current User:	git
Using RVM:	no
Ruby Version:	2.1.5p273
Gem Version:	2.2.2
Bundler Version:1.7.9
Rake Version:	10.3.2
Sidekiq Version:2.17.0

GitLab information
Version:	7.5.3
Revision:	fatal: Not a git repository (or any parent up to mount point /raid)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Directory:	/raid/git/gitlab/gitlab
DB Adapter:	mysql2
URL:		http://myserver/gitlab
HTTP Clone URL:	http://myserver/gitlab/some-project.git
SSH Clone URL:	git@myserver:some-project.git
Using LDAP:	no
Using Omniauth:	no

GitLab Shell
Version:	2.2.0
Repositories:	/raid/git/repos/
Hooks:		/raid/git/gitlab/gitlab-shell/hooks/
Git:		/usr/bin/git
```

Gitlab version 7.5.3 (7-5-stable)
Latest commit: b656b8592fc67b466ae2dd3b99ad35d663721b37

**Comment by Razer6** [comment]
Thanks for the issue report. This issue has already been fixed in newer versions of GitLab. Due to the size of this project and our limited resources we are only able to support the latest stable release as outlined in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#issue-tracker). In order to get this bug fix and enjoy many new features please [upgrade](https://gitlab.com/gitlab-org/gitlab-ce/tree/master/doc/update). If you still experience issues at that time please open a new issue following our issue tracker guidelines found in the [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#issue-tracker-guidelines).

**Comment by dloghin** [comment]
Hi,

Meanwhile, I have updated my gitlab to the latest version, 8.2.1. I have encountered the same issue as described in my previous comment. However, I managed to solve it using the following workaround: I have manually updated all 'iid's in the 'issues' table. When I have previously updated to version 7.5.0, the 'iid' values where set to the default (NULL), but it seems that these values lead to the described display problem. 

I observed that the first issue for a project has the 'iid' 1. Hence, for a my first old issue ('id' = 1) I have updated its 'iid' to 1 using MYSQL command:

mysql> update issues set iid=1 where id=1;

Hope this can help others users that are updating their old gitlab installations.
